### PR TITLE
#8 - removing hard coded data

### LINKED ISSUE #8: Removing hard-coded data
The aim of this issue is to change hard-coded data like date, meetup name, speakers etc., so that they are downloaded from json files.

### PR DESCRIPTION
**Pull request by paulinaszkudlarek**
This should close #8.
image
image
image
image
image

### CODE REVIEW

File: frontend/components/Landing/AboutMeetup.vue [hunk 1]
```diff
@@ -38,10 +38,12 @@
...
       </div>
-      <h3 class="block text-center text-lg font-semibold text-laravel -mb-8">Organizatorzy</h3>
-      <div class="block justify-center md:flex md:w-3/5 md:m-auto">
-        <img class="mx-auto center h-48"  alt="logo Blumilk" src="/images/companies/blumilk.webp"/>
-        <img class="mx-auto center h-48 object-cover scale-140" alt="logo Highsolutions" src="/images/companies/highsolutions.webp"/>
+      <div>
+        <h3 class="block text-center text-lg font-semibold text-laravel -mb-8">Organizatorzy</h3>
+        <div class="block justify-center md:flex md:w-3/5 md:m-auto">
+          <img class="mx-auto center h-48"  alt="logo Blumilk" src="/images/companies/blumilk.webp"/>
+          <img class="mx-auto center h-48 object-cover scale-140" alt="logo Highsolutions" src="/images/companies/highsolutions.webp"/>
```
Comment (EwelinaSkrzypacz) [inline frontend/components/Landing/AboutMeetup.vue]:
```suggestion
          <img class="mx-auto center h-48 object-cover scale-140" alt="logo HighSolutions" src="/images/companies/highsolutions.webp"/>
```

File: frontend/components/Landing/CountDown.vue [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+<template>
+  <h2 class="uppercase font-semibold text-3xl text-zinc-300" v-if="finish == false">
+    Następny meetup już
+    <span class="text-laravel" v-if="isToday"> dziś!</span>
+    <span class="text-laravel" v-else-if="isTomorrow"> jutro!</span>
+    <span class="text-laravel" v-else > za {{days}} dni!</span>
+  </h2>
+  <div class="mt-3 space-y-3 sm:tracking-wider text-white" v-else>
+    <h1 class="text-3xl font-bold sm:text-4xl">Już niedługo ogłosimy datę kolejnego <span class="text-laravel">meetupu!</span></h1>
+    <p class="text-3xl">Obserwuj nasz profil na <a href="https://www.facebook.com/laravelpolandmeetup/" class="text-cyan-600 tracking-normal font-bold">facebooku</a>, aby być na bieżąco!</p>
```
Comment (EwelinaSkrzypacz) [inline frontend/components/Landing/CountDown.vue]:
```suggestion
    <p class="text-3xl">Obserwuj nasz profil na <a href="https://www.facebook.com/laravelpolandmeetup/" class="text-cyan-600 tracking-normal font-bold">Facebooku</a>, aby być na bieżąco!</p>
```

File: frontend/components/Landing/CountDown.vue [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    <p class="text-2xl font-semibold">
+      Sprawdź też, czy nie ominęło cię coś interesującego w
```
Comment (EwelinaSkrzypacz) [inline frontend/components/Landing/CountDown.vue]:
```suggestion
      Sprawdź też, czy nie ominęło Cię coś interesującego w
```

**Review by EwelinaSkrzypacz — COMMENTED** [review]
1. My impression is that here too much information is highlighted :
image
Maybe we should make another color font and size for "Obserwuj nasz profil na [facebooku](https://www.facebook.com/laravelpolandmeetup/), aby być na bieżąco!"
It will be great if this will be consistent with this:
image

File: frontend/components/Landing/CountDown.vue [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+let isTomorrow = false;
+
+let now = new Date();
+let endDate = new Date(props.date);
+
+if (endDate.getDate()  == now.getDate()  && 
+    endDate.getMonth() == now.getMonth() &&
+    endDate.getYear()  == now.getYear()) {
+    isToday = true;
+} else if (
```
Comment (kamilpiech97) [inline frontend/components/Landing/CountDown.vue]:
```suggestion
if (endDate.toDateString() === now.toDateString()) {
    isToday = true;
} else if (
```
Maybe that its more readable than three conditional if statement?

**Review by kamilpiech97 — CHANGES_REQUESTED** [review]

File: frontend/components/Landing/CountDown.vue (hunk 4 above)
Comment (paulinaszkudlarek) [reply frontend/components/Landing/CountDown.vue]: this is much better :) i will change it

File: frontend/components/Landing/CountDown.vue [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+let now = new Date();
+let endDate = new Date(props.date);
+
+if (endDate.toDateString() === now.toDateString()) {
+    isToday = true;
+} else if (
+    endDate.getDate()  == now.getDate() +1 && 
+    endDate.getMonth() == now.getMonth()   &&
+    endDate.getYear()  == now.getYear()) {
+    isTomorrow = true;
```
Comment (kamilpiech97) [inline frontend/components/Landing/CountDown.vue]: What about this? :thinking:

**Review by kamilpiech97 — CHANGES_REQUESTED** [review]

File: frontend/components/Landing/CountDown.vue (hunk 5 above)
Comment (paulinaszkudlarek) [reply frontend/components/Landing/CountDown.vue]: ok, changed:)

**Review by kamilpiech97 — DISMISSED** [review]

File: frontend/components/Landing/CountDown.vue [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  if( timeLeft <= 0) {
+    finish = true;
+  }
+  days = Math.floor(timeLeft / 86400); 
+}
+countDown();
+</script>
+
+<template>
+  <h2 class="uppercase font-semibold text-3xl text-zinc-300" v-if="finish == false">
```
Comment (krzysztofrewak) [inline frontend/components/Landing/CountDown.vue]:
```suggestion
  <h2 class="uppercase font-semibold text-3xl text-zinc-300" v-if="!finish">
```

Using `==` is a bad practice in most cases. Negation would be fine here.

File: frontend/components/Landing/Counters.vue [hunk 7]
```diff
@@ -1,5 +1,39 @@
...
+  let speakers = []; 
+
+  for(let meetup of data.value) {
+    for(let presentation of meetup.presentations){
+      for(let speaker of presentation.speakers){
+        speakers.push(speaker.name);
+      }
+    }
+  }
+  let uniqueSpeakers = [...new Set(speakers)].length;
```
Comment (krzysztofrewak) [inline frontend/components/Landing/Counters.vue]:
```suggestion
  let uniqueSpeakers = speakers.length;
```

Whhy not?

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]
What do you wthink about producing some `counters.json` or `statistics.json` that would present already-computed data in smaller file?

**Comment by krzysztofrewak** [comment]
There are none or some spaces between some sections on main page. I would prefer to unify this.

image

File: frontend/components/Landing/Counters.vue (hunk 7 above)
Comment (paulinaszkudlarek) [reply frontend/components/Landing/Counters.vue]: 'Set' let me store unique values, if I remove it I will have doubled speakers, but.... I will change the logic and just count speakers from people.json

File: frontend/components/Landing/Counters.vue [hunk 8]
```diff
@@ -1,5 +1,17 @@
...
+
+const data = ref({})
+
+
+onMounted( () => {
+  fetch('/api/counters.json').then((response) => response.json()).then((counters) => {
+    data.value = counters
+  })
+})
+console.log('data', data.value)
```
Comment (krzysztofrewak) [inline frontend/components/Landing/Counters.vue]:
```suggestion
```

File: resources/people.json [hunk 9]
```diff
@@ -83,16 +83,16 @@
...
+    "image": "../public/images/speakers/marcintracz.webp", 
     "bio": "#",
-    "linkedin":  null
+    "linkedin":  "https://www.linkedin.com/in/marcin-tracz-2a3307139/"
   },
   "@mateuszgarbarczyk": {
-    "name": "MateuszGarbarczyk",
-    "company": "HIGHSolutions",
+    "name": "Mateusz Garbarczyk",
+    "company": "MirIT",
```
Comment (krzysztofrewak) [inline resources/people.json]: Hmm, I think that company should not be a part of a person. People tend to change companie frequently. ;)

Comment (krzysztofrewak) [reply resources/people.json]: Every presentation has a Speaker object that is built from a person and a company.

File: resources/companies.json [hunk 10]
```diff
@@ -25,5 +25,10 @@
     "name": "Trui",
     "logo": "/images/companies/trui.png",
     "location": "Poznań"
+  },
+  "@mirit": {
+    "name": "MirIT",
+    "logo": "#",
+    "location": "#"
```
Comment (krzysztofrewak) [inline resources/companies.json]:
```suggestion
    "location": "Pleszew"
```

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

File: resources/people.json (hunk 9 above)
Comment (paulinaszkudlarek) [reply resources/people.json]: that's right, company will be removed from resources/people.json

File: frontend/components/Landing/Counters.vue (hunk 8 above)
Comment (paulinaszkudlarek) [reply frontend/components/Landing/Counters.vue]: ups :)

**Review by krzysztofrewak — DISMISSED** [review]
There are a lot of syntax problems here, but I will set up eslint properly in next PR.

**Review by EwelinaSkrzypacz — APPROVED** [review]

**Review by krzysztofrewak — APPROVED** [review]

**Review by kamilpiech97 — APPROVED** [review]

**Pull request merged**